bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/clangd/ClangdOptionsDefaults.java

```diff
@@ -0,0 +1,61 @@
+package org.eclipse.cdt.lsp.clangd;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.cdt.utils.PathUtil;
+import org.eclipse.core.runtime.IPath;
+
+public class ClangdOptionsDefaults implements ClangdOptions {
+
+	@Override
+	public boolean preferClangd() {
+		return false;
+	}
+
+	@Override
+	public String clangdPath() {
+		return Optional.ofNullable(PathUtil.findProgramLocation("clangd", null)) //$NON-NLS-1$
+				.map(IPath::toOSString)//
+				.orElse("clangd"); //  //$NON-NLS-1$
+	}
+
+	@Override
+	public boolean useTidy() {
+		return true;
+	}
+
+	@Override
+	public boolean useBackgroundIndex() {
+		return true;
+	}
+
+	@Override
+	public String completionStyle() {
+		return "detailed";
+	}
+
+	@Override
+	public boolean prettyPrint() {
+		return true;
+	}
+
+	@Override
+	public String queryDriver() {
+		return Optional.ofNullable(PathUtil.findProgramLocation("gcc", null)) //$NON-NLS-1$
+				.map(p -> p.removeLastSegments(1).append(IPath.SEPARATOR + "*"))// //$NON-NLS-1$
+				.map(IPath::toString)//
+				.orElse(""); //  //$NON-NLS-1$
+	}
+
+	@Override
+	public List<String> additionalOptions() {
+		return Collections.emptyList();
+	}
+
+	@Override
+	public List<String> toList() {
+		throw new RuntimeException("This method is not intended to be called by clients"); //$NON-NLS-1$
```

Comment:
It was intended. If it is not we need to remove it from interface.

Reply:
One might argue that the toList() method shouldn't be in the interface, however I didn't want to change this. 
On the other side it makes sense for the defaults to implement this interface, so when a new option is added, the class which implements the defaults gets a compile error. Having the toList() method in the defaults doesn't make sense to me though, that's why I added the exception there.

So IMO the clean solution would be to extract the `toList()` method, what do you think?

Reply:
> clean solution would be to extract the toList()

We have a problem because we are reusing interface `ClangdOptions` for a different purpose. This interface was not designed to represent default values. We can rework it to serve us for the case of default values as well.